sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -245,19 +247,26 @@ internal class QProductCenterManager internal constructor(
     ) {
         loadProducts(object : QonversionProductsCallback {
             override fun onSuccess(products: Map<String, QProduct>) {
-                val storeIds = products.map { it.value.skuDetail?.sku }.filterNotNull()
-                repository.eligibilityForProductIds(
-                    storeIds,
-                    installDate,
-                    object : QonversionEligibilityCallback {
-                        override fun onSuccess(eligibilities: Map<String, QEligibility>) {
-                            val resultForRequiredProductIds =
-                                eligibilities.filter { it.key in productIds }
-                            callback.onSuccess(resultForRequiredProductIds)
-                        }
+                val result: MutableMap<String, QEligibility> = mutableMapOf()
+                products.forEach {
+                    val hasTrialOrIntroOffer = it.value.storeDetails?.hasTrialOrIntroOffer ?: false
+                    val isInApp = it.value.storeDetails?.productType == QProductType.InApp
```

Comment:
let's maybe create an utilitar bool property for this? I think there would be lots of such checks